src/Akka.Persistence.Linq2Db.TagTableMigration/Options.cs

```diff
@@ -0,0 +1,113 @@
+﻿// -----------------------------------------------------------------------
+//  <copyright file="Options.cs" company="Akka.NET Project">
+//      Copyright (C) 2013-2023 .NET Foundation <https://github.com/akkadotnet/akka.net>
+//  </copyright>
+// -----------------------------------------------------------------------
+
+using Akka.Configuration;
+using Akka.Persistence.Sql.Linq2Db;
+
+namespace Akka.Persistence.Linq2Db.TagTableMigration;
+
+public enum DatabaseType
+{
+    SqlServer,
+    Sqlite,
+    PostgreSql,
+    MySql
+}
+
+public enum ProviderType
+{
+    SqlServer,
+    SqlServer2000,
+    SqlServer2005,
+    SqlServer2008,
+    SqlServer2012,
+    SqlServer2014,
+    SqlServer2016,
+    SqlServer2017,
+    MySql,
+    MySqlOfficial,
+    MySqlConnector,
+    PostgreSql,
+    PostgreSql92,
+    PostgreSql93,
+    PostgreSql95,
+    Sqlite,
+    SqliteClassic,
+    SqliteMs
+}
+
+public static class Extensions
+{
+    public static string ToHocon(this DatabaseType db)
+        => db switch
+        {
+            DatabaseType.SqlServer => "sql-server",
+            DatabaseType.Sqlite => "sqlite",
+            DatabaseType.PostgreSql => "postgresql",
+            DatabaseType.MySql => "mysql",
+            _ => throw new ArgumentException($"Unknown database type: {db}")
+        };
+
+    public static string ToHocon(this ProviderType type)
```

Comment:
Boilerplate for the CLI args --> Linq2Db configuration pipeline by the looks of it